### DISCUSSION TITLE
Custom equalizer settings

### DISCUSSION THREAD

**Discussion by Michsior14** [Sapd/HeadsetControl#253] [Ideas]
Some time ago I've implemented equalizer presets and thought about adding custom equalizer settings (eg. via `e` param).

I could easily use `get_data_from_parameter` from `dev.c` to get the content of buffer that would represents it, but how to handle the different parameter type in `handle_feature` (`int param`)? Just by introducing another one that would be optional `char* stringParam` or struct holding different types? How you would like to handle it @Sapd?

**Comment by Sapd** [comment]
In fact I did not thought about that when writing the function.

I think best way would be to change the parameter of `handle_feature` from `int param` to `void* param`  and call it with ie. `&sidetone_loudness` then cast it depending on the case in `handle_feature` .

It also might make sense to parse the equalizer parameter directly in main.c with something like `get_data_from_parameter` and pass it with such a struct to the `device_found->`
```
// defined in device.h
struct equalizer_settings {
    int size;
    int equalizer_parameter_list[];
};
```
( https://en.wikipedia.org/wiki/Flexible_array_member )

If you want to use `get_data_from_parameter` it might make sense to move it from dev.c/h to utility.c/h
